frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/EmbeddingOption.module.css

```diff
@@ -0,0 +1,25 @@
+.Primary {
+  fill: var(--mb-color-icon-primary);
+}
```

Comment:
This is like atomic classes. It will be a lot more complicated if we want to override more properties with the same colors.

I'd prefer my previous solution where we define new ad-hoc colors e.g.
```css
  --mb-color-adhoc-primary: var(--mb-color-icon-primary);
  --mb-color-adhoc-secondary: var(--mb-color-icon-secondary)
```

then you could override these same 2 colors under `.disabled` with different semantic colors.

```css
.disabled {
  --mb-color-adhoc-primary: var(--mb-color-icon-primary-disabled);
  --mb-color-adhoc-secondary: var(--mb-color-icon-secondary-disabled)
}
```

then we could hard code only these 2 ad-hoc colors to all SVGs.

This approach would scale better with more CSS colors with multiple properties (fill, stroke, etc.)

Nonetheless, they achieve the same result. I'm not strongly against this approach here, but it won't be easy to reason about when the colors are more complex.

Reply:
One thing I found with this approach is `.primary` (excuse me for not capitalizing since I'm too lazy) and `.secondaryFill` apply the same property have different name patterns. This reeks code smell in my opinion.

Reply:
how would we hardcode the colors into the component? just using `var(--whatever)` directly in `fill` and `stroke`?

Reply:
yep, here's [the example](https://github.com/metabase/metabase/blob/64f130cc7f090918c20c0d242d7d8660b9c1f0c7/frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/InteractiveEmbedding.svg?short_path=4fd8888). Don't mind the `<style>` tag though, it's for the preview in code editors.

Reply:
oh ok nice. I'll just do the change now since it's not set in stone

Reply:
The problem with not using base colors here is that we are forced to define 2 more colors upfront `icon-primary-disabled` and `icon-secondary-disabled`, had we allow base colors on files other than `colors.module.css` we could hard-code just 2 colors `icon-primary` and `icon-secondary` on all SVGs and override these 2 CSS custom properties instead when the `.disabled` class is applied.

To be clear, this is what I did in the example I linked above.

Reply:
I just want to point out that it might also be a good idea to define 4 colors upfront though. If we think that having these 2 additional colors suits our design system better. The problem is that I still don't know what concrete criteria we should use to determine if we need new semantic colors for our design system.

Reply:
The first question should definitely be "will this be used outside of this context"

Reply:
since this is the very first location, I think not yet. But I imagine all icons should roughly have some design system to back them otherwise colors would be a mess wouldn't it?

So in that sense I think these colors are definitely being used outside this context (embedding option card SVG colors)

Reply:
it does make sense - if we are using icons with a brand color and they can be disabled, we really should be using the same colors for everything. I think it does make sense to add global semantic colors for these since we can sometimes have two-tone enabled (with brand color) and disabled (with gray) icons

Reply:
It would jsut be a

Reply:
Ok, we're somewhat on the same page. I think allowing the use of base colors outside `colors.module.css` would encourage more exploration like this, and this allows the pattern to be established before we put that back in `colors.module.css`.

However this could create a situation where we don't clean up these ad-hoc colors, and they're not really well maintained. Imagine a few more places that uses ad-hoc colors that have the same meaning, in that case those colors shouldn't be ad-hoc anymore and should live inside `colors.module.css`

On the other hand, forcing everything to be defined inside `colors.module.css` could be a good idea since it forces us to think systematically about the color we want to use. The problem is the growing semantic colors we'll be using though. I would imagine there could be a lot more semantic colors that are only used in a handful of places that could potentially pollute `colors.module.css`.

to be fiar, I don't see a clear winner, it depends on what kind of situations we're encouraging.

Reply:
I think I agree with keeping things in `colors.module.css`. One of the main benefits a design system brings is (hopefully) thoughtful restriction. Just like with spacing, if we're forced to use the semantic colors from `colors.module.css` we'll ensure that we're not accidentally expanding our color palette and we'll (again hopefully) be more thoughtful about which colors should go where

Reply:
Alright! it's decided then. @iethree  also thinks [we should keep things inside `colors.module.css`](https://metaboat.slack.com/archives/C057WD5L0JG/p1730732847564649?thread_ts=1730317945.468619&cid=C057WD5L0JG)

Reply:
cool! I've pushed a fix with your suggestion of changing CSS vars instead of classes, and then I've added 4 semantic colors for the icons